src/advanced_inv.cpp

```diff
@@ -454,41 +486,67 @@ void advanced_inventory::menu_square( uimenu *menu )
 
 inline char advanced_inventory::get_location_key( aim_location area )
 {
+    ;
     switch( area ) {
         case AIM_INVENTORY:
             return 'I';
         case AIM_WORN:
             return 'W';
-        case AIM_SOUTHWEST:
-            return '1';
-        case AIM_SOUTH:
-            return '2';
-        case AIM_SOUTHEAST:
-            return '3';
-        case AIM_WEST:
-            return '4';
         case AIM_CENTER:
             return '5';
-        case AIM_EAST:
-            return '6';
-        case AIM_NORTHWEST:
-            return '7';
-        case AIM_NORTH:
-            return '8';
-        case AIM_NORTHEAST:
-            return '9';
         case AIM_ALL:
             return 'A';
         case AIM_DRAGGED:
             return 'D';
         case AIM_CONTAINER:
             return 'C';
+        case AIM_NORTH:
+        case AIM_SOUTH:
+        case AIM_EAST:
+        case AIM_WEST:
+        case AIM_NORTHEAST:
+        case AIM_NORTHWEST:
+        case AIM_SOUTHEAST:
+        case AIM_SOUTHWEST:
+            {
+                return std::to_string( get_direction_key( area ) ).c_str()[0];
```

Comment:
Because other parts of the code need the direction key as an int - specifically uimenu.addentry() - and duplicating them as literals seemed like bad practice.